test/net/sourceforge/kolmafia/objectpool/ConcoctionTest.java

```diff
@@ -2,24 +2,46 @@
 
 import static org.junit.jupiter.api.Assertions.*;
 
+import java.util.HashMap;
+import java.util.Map;
 import net.java.dev.spellcast.utilities.LockableListModel;
 import net.sourceforge.kolmafia.persistence.ConcoctionDatabase;
-import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 
 /* This test was triggered by a runtime error traced back to sorting usable concoctions that
-said "Comparison method violates its general contract!"  It is likely that the error was caused by
-dynamic data, i.e. something that changed because a character left Ronin, but it can't hurt to verify
-the comparison against static data.
+said "Comparison method violates its general contract!"
  */
 
 public class ConcoctionTest {
 
+  static final Map<Integer, Concoction> idsAndMaps = new HashMap<>();
+  static int maxIndex;
+  static String msg;
```

Comment:
I don't see why msg / maxIndex have to be static.

You're not saving any allocations given that you're constructing this string from scratch every time, and maxIndex is basically a glorified idsAndMaps.size().

Reply:
msg doesn't need to be static.  maxIndex is used in buildIndexAndData which has to be static to be @BeforeAll and code didn't compile without it.  In Ye Olden Days idsAndMaps.size() was an object reference and a function call which was more expensive than accessing a field.  I don't know whether that is true for 21st Century Java but it is habit from a time and place where it was less efficient.